File: q11/04.swift
#!/usr/bin/swift

import Foundation

var a = 1
var b = 1
var count = 0

while count < 11 {
    var c = a + b
    var sum = 0
    var characters = String(c).characters
    for character in characters {
        let number = Int(String(character))!
        sum += number
    }
    if c % sum == 0 {
        print(c)
        count += 1
    }
    a = b
    b = c
}

File: q11/04.rb
#!/usr/bin/env ruby

a = 1
b = 1
count = 0

while count < 11
  c = a + b
  sum = 0
  c.to_s.split(//).each { |e| sum += e.to_i }
  if (c % sum).zero?
    puts c
    count += 1
  end
  a, b = b, c
end
